components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java

```diff
@@ -262,6 +262,12 @@ public Response unRegister(@QueryParam("applicationName") String applicationName
      * @return true if user has any of create, publish or subscribe permissions
      */
     private boolean isUserAccessAllowed(String username) {
+        String authScheme = (String) securityContext.get(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME);
```

Comment:
Let's add a comment mentioning, 
`The legacy permission validation (publish, subscribe, create) is skipped when the authentication is OAuth (Opaque or JWT). There's already a scope validation engaged there so no need of another permission check.`

Reply:
Fixed